node/src/components/block_validator.rs

```diff
@@ -149,11 +149,20 @@ where
                 responder,
                 block_timestamp,
             }) => {
-                let block_deploys = block
-                    .deploys()
+                let block_deploys = block.deploys();
+                let deploy_count = block_deploys.len();
+                // Collect the deploys in a set; this also deduplicates them.
+                let block_deploys: HashSet<_> = block_deploys
                     .iter()
                     .map(|deploy_hash| **deploy_hash)
-                    .collect::<HashSet<_>>();
+                    .collect();
+                if block_deploys.len() != deploy_count {
+                    info!(
+                        deploys = ?block.deploys(), ?sender,
```

Comment:
really, we're logging 2600 deploy hashes? or even whole bodies?

Reply:
If we receive a block with an attempted replay attack? Why not?

Reply:
On the other hand: The deploy hashes won't actually tell us much; we should rather log the block proposer maybe?

Reply:
My question is _why_?

Reply:
I would say that we should log the block proposer and replayed hashes. This shouldn't happen often so we can afford finding duplicates.

Reply:
Turns out we only have a `BlockLike` there, not a `Block`, so no proposer etc.
Maybe we should move this check into consensus after all?

Reply:
As discussed, I changed the log to only list the duplicated deploys.